account-kit/infra/src/middleware/gasManager.ts

```diff
@@ -188,6 +205,107 @@ export function alchemyGasAndPaymasterAndDataMiddleware(
           : {}),
       });
 
+      let erc20Context:
+        | {
+            tokenAddress: Address;
+            maxTokenAmount: bigint;
+            permit?: Hex;
+          }
+        | undefined = undefined;
+      if (policyToken !== undefined) {
+        let maxAmountToken = policyToken.maxTokenAmount
+          ? policyToken.maxTokenAmount
+          : MaxUint256;
+
+        erc20Context = {
+          tokenAddress: policyToken.address,
+          maxTokenAmount: maxAmountToken,
+        };
+        if (policyToken.approvalMode === "PERMIT") {
+          // get a paymaster address
+          let paymasterAddress = "0x";
+          const paymasterData = await (
+            client as AlchemySmartAccountClient
+          ).request({
+            method: "pm_getPaymasterStubData",
+            params: [
+              userOp,
+              account.getEntryPoint().address,
+              toHex(client.chain.id),
+              {
+                policyId: Array.isArray(policyId) ? policyId[0] : policyId,
+              },
+            ],
+          });
+
+          paymasterAddress = paymasterData.paymaster
+            ? paymasterData.paymaster
+            : paymasterData.paymasterAndData
+            ? paymasterData.paymasterAndData.slice(0, 42)
+            : "0x";
```

Comment:
Setting the address to "0x" in the absent case is a little odd because "0x" is not an address (this would be an invalid input in the Rundler, for example). Consider using `undefined` to represent this case instead. (If existing APIs require this to be "0x", then ignore this, but also we should change the APIs in that case).

If you do that, watch out for the case where `paymasterAndData` is `"0x"`, in which you'll need to make sure to set the address to undefined.

Reply:
added a check instead of swallow the error